gears/src/x/ante.rs

```diff
@@ -450,6 +451,60 @@ impl<
                         account_number: acct.get_account_number(),
                     }
                     .encode_to_vec(),
+                    SignMode::LegacyAminoJson => {
+                        let mut msgs = vec![];
+                        for msg in tx.tx.get_msgs() {
+                            // TODO: don't have proper value getter, maybe extend trait
+                            let mut value: serde_json::Map<String, serde_json::Value> =
+                                serde_json::from_slice(
+                                    &serde_json::to_vec(&msg)
+                                        .map_err(|_| AnteError::LegacyAminoJson)?,
```

Comment:
We could do with a better solution here... maybe we have a `AminoSerde` trait, kind of like our `ValueRenderer` trait

Reply:
Your comment is related to my `TODO`:                            
```rs
   // TODO: don't have proper value getter, maybe extend trait
```

You're right, it's better to add some proper solution for serializing domain tx types into amino types.